pymongo/collection.py

```diff
@@ -242,32 +262,37 @@ def __init__(
             else:
                 self.__create(name, kwargs, collation, session)
 
-    def _socket_for_reads(self, session):
+    def _socket_for_reads(
+        self, session: _ServerMode
+    ) -> Iterator[Tuple[SocketInfo, Union[PrimaryPreferred, Primary]]]:
         return self.__database.client._socket_for_reads(self._read_preference_for(session), session)
 
-    def _socket_for_writes(self, session):
+    def _socket_for_writes(
+        self, session: Optional[ClientSession]
+    ) -> _GeneratorContextManager[SocketInfo]:
         return self.__database.client._socket_for_writes(session)
 
     def _command(
         self,
-        sock_info,
-        command,
-        read_preference=None,
-        codec_options=None,
-        check=True,
-        allowable_errors=None,
-        read_concern=None,
-        write_concern=None,
-        collation=None,
-        session=None,
-        retryable_write=False,
-        user_fields=None,
-    ):
+        sock_info: SocketInfo,
+        command: Mapping[str, Any],
+        read_preference: Optional[_ServerMode] = None,
+        codec_options: Optional[CodecOptions] = None,
+        check: bool = True,
+        allowable_errors: Optional[Container[Any]] = None,
+        read_concern: Optional[ReadConcern] = None,
+        write_concern: Optional[WriteConcern] = None,
+        collation: Union[Collation, Mapping[str, Any], None] = None,
```

Comment:
We have an alias for this: [`_CollationIn`](https://github.com/mongodb/mongo-python-driver/blob/94fabf5e988b02041693ff7f0921e4eee32ee9a0/pymongo/typings.py#L26) (for all usages in this file).